src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,41 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents the Contact data type in the address book. 
+ * Is superclass of any user details. 
+ */
+public class Contact {
+    protected String value;
+    protected boolean isPrivate;
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        throw new UnsupportedOperationException("This method is to be implemented by child classes");
```

Comment:
With this implementation, would it be better to make the class abstract and let this method be abstract?